### PR TITLE
Task-6: change detection

### PR DESCRIPTION
**Pull request by homergit**

### CODE REVIEW

File: src/app/features/add-course/add-course-window/add-course-window.component.html [hunk 1]
```diff
@@ -0,0 +1,48 @@
+<div class="add-course">
+  <h1 class="add-course__title">New Course</h1>
+
+  <form [formGroup]="courseForm" #registerForm="ngForm" (ngSubmit)="submitCourse" class="add-course__form">
```
Comment (torin4fan) [inline src/app/features/add-course/add-course-window/add-course-window.component.html]: (ngSubmit)="submitCourse" doesn't work. Use (ngSubmit)="submitCourse()" instead

File: src/app/features/add-course/add-course-window/add-course-window.component.html [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    </label>
+
+    <app-duration [formGroup]="courseForm"></app-duration>
+    <app-date [formGroup]="courseForm"></app-date>
+    <app-authors [formGroup]="courseForm"></app-authors>
+
+    <div class="add-course__actions">
+      <button
+        [disabled]="courseForm.invalid"
+        (click)="submitCourse()"
```
Comment (torin4fan) [inline src/app/features/add-course/add-course-window/add-course-window.component.html]: You're already have (ngSubmit)="submitCourse". The (click)="submitCourse()" is redundant. Hust add type='submit' to the button instead

File: src/app/features/add-course/add-course-window/add-course-window.component.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    this.courseForm.valueChanges.subscribe(value => {
+      this.formValue = value;
+    });
+  }
+
+  get f() { return this.courseForm.controls; }
+
+  submitCourse() {
+    this.course = {
+      id: 12,
```
Comment (torin4fan) [inline src/app/features/add-course/add-course-window/add-course-window.component.ts]: You could generate id from js time method.

File: src/app/features/course/courses-section/courses-section.component.ts [hunk 4]
```diff
@@ -4,8 +4,8 @@ import {MatDialog} from '@angular/material/dialog';
 import {Course, DeletedItem} from '../../../core/models/course';
 import {FilterPipe} from '../../../core/pipes/filter.pipe';
 import {OrderByPipe} from '../../../core/pipes/order-by.pipe';
-import {CoursesService} from "../courses.service";
-import {DialogComponent} from "../../../shared/dialog/dialog.component";
+import {CoursesService} from '../courses.service';
+import {DialogComponent} from '../../../shared/dialog/dialog.component';
 
 @Component({
```
Comment (torin4fan) [inline src/app/features/course/courses-section/courses-section.component.ts:10]: add OnPush strategy to this component. Would be great to add this strategy to each component in app.

File: src/app/features/add-course/add-course-window/add-course-window.component.html (hunk 1 above)
Comment (homergit) [reply src/app/features/add-course/add-course-window/add-course-window.component.html]: I was inattentive, thank you!

File: src/app/features/add-course/add-course-window/add-course-window.component.html (hunk 2 above)
Comment (homergit) [reply src/app/features/add-course/add-course-window/add-course-window.component.html]: removed, thanks!

File: src/app/features/add-course/add-course-window/add-course-window.component.ts (hunk 3 above)
Comment (homergit) [reply src/app/features/add-course/add-course-window/add-course-window.component.ts]: I thought to do it in the http module in task 8, but thanks :)

File: src/app/features/course/courses-section/courses-section.component.ts (hunk 4 above)
Comment (homergit) [reply src/app/features/course/courses-section/courses-section.component.ts:10]: done

**Pull request merged**